.dotnet/src/Custom/Images/ImageClient.Protocol.cs

```diff
@@ -16,4 +18,180 @@ public virtual ClientResult GenerateImage(BinaryContent content, RequestOptions
+    /// </item>
+    /// <item>
+    /// <description>
+    /// Please try the simpler <see cref="GenerateImageEditsAsync(BinaryData, string, string, int?, ImageEditOptions)"/> or <see cref="GenerateImageEditsAsync(Stream, string, string, int?, ImageEditOptions)"/> convenience overload with strongly typed models first.
+    /// </description>
+    /// </item>
+    /// </list>
+    /// </summary>
+    /// <param name="content"> The content to send as the body of the request. </param>
+    /// <param name="contentType"> The content type of the request. </param>
+    /// <param name="options"> The request options, which can override default behaviors of the client pipeline on a per-call basis. </param>
+    /// <exception cref="ArgumentNullException"> <paramref name="content"/> is null. </exception>
+    /// <exception cref="ArgumentException"> <paramref name="contentType"/> is an empty string, and was expected to be non-empty. </exception>
+    /// <exception cref="ClientResultException"> Service returned a non-success status code. </exception>
+    /// <returns> The response returned from the service. </returns>
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public virtual async Task<ClientResult> GenerateImageEditsAsync(BinaryContent content, string contentType, RequestOptions options = null)
+    {
+        Argument.AssertNotNull(content, nameof(content));
+        Argument.AssertNotNullOrEmpty(contentType, nameof(contentType));
+
+        options ??= new RequestOptions();
+
+        using PipelineMessage message = CreateCreateImageEditsRequest(content, contentType, options);
+
+        Shim.Pipeline.Send(message);
+
+        PipelineResponse response = message.Response!;
+
+        if (response.IsError && options.ErrorOptions == ClientErrorBehaviors.Default)
+        {
+            throw await ClientResultException.CreateAsync(response).ConfigureAwait(false);
+        }
+
+        return ClientResult.FromResponse(response);
+    }
+
+    /// <summary>
+    /// [Protocol Method] Creates an edited or extended image given an original image and a prompt.
+    /// <list type="bullet">
+    /// <item>
+    /// <description>
+    /// This <see href="https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/samples/ProtocolMethods.md">protocol method</see> allows explicit creation of the request and processing of the response for advanced scenarios.
+    /// </description>
+    /// </item>
+    /// <item>
+    /// <description>
+    /// Please try the simpler <see cref="GenerateImageVariations(BinaryData, string, int?, ImageVariationOptions)"/> or <see cref="GenerateImageVariations(Stream, string, int?, ImageVariationOptions)"/> convenience overload with strongly typed models first.
+    /// </description>
+    /// </item>
+    /// </list>
+    /// </summary>
+    /// <param name="content"> The content to send as the body of the request. </param>
+    /// <param name="contentType"> The content type of the request. </param>
```

Comment:
We may want to add that this should be multipart/form-data, including the boundary used to compose the <paramref name="content"/> value.  It might make sense to link to the RFC as well, which is here: https://www.rfc-editor.org/rfc/rfc7578, or suggest that people use https://learn.microsoft.com/en-us/dotnet/api/system.net.http.multipartformdatacontent?view=net-8.0 to compose this correctly.